rust/src/fvm/externs.rs

```diff
@@ -99,17 +99,19 @@ impl Consensus for CgoExterns {
                 &mut miner_id,
                 &mut epoch,
                 &mut fault_type,
+                &mut gas_used,
             ) {
-                0 => Ok(Some(ConsensusFault {
-                    target: Address::new_id(miner_id),
+                0 => Ok(ConsensusFaultWithGas {
```

Comment:
Personally, I'd return a tuple instead of a new type.

Also, this was clearly broken before as we had no case that returned `None`, but I don't think that having a "no fault" variant is the right fix. See https://github.com/filecoin-project/ref-fvm/pull/330#discussion_r807523704.